src/datachain/lib/pytorch.py

```diff
@@ -88,7 +88,9 @@ def __iter__(self) -> Iterator[Any]:
         if self.catalog is None:
             self.catalog = self._get_catalog()
         total_rank, total_workers = self.get_rank_and_workers()
-        ds = DataChain(name=self.name, version=self.version, catalog=self.catalog)
+        ds = DataChain._create(
```

Comment:
why is it not complaining about accessing the protected method? can we avoid it here?